src/test/java/org/mastodon/util/ColorUtilsTest.java

```diff
@@ -29,10 +30,31 @@ public void testGetMixedColor()
 	}
 
 	@Test
-	public void testScaleAlpa()
+	public void testScaleAlpha()
 	{
 		assertEquals( 0x00ffffff, ColorUtils.scaleAlpha( 0x11ffffff, -1 ) );
 		assertEquals( 0x22aabbcc, ColorUtils.scaleAlpha( 0x44aabbcc, 0.5f ) );
 		assertEquals( 0x11ffffff, ColorUtils.scaleAlpha( 0x11ffffff, 2 ) );
+
+		Color input = Color.BLACK;
+		float ratio = 0.5f;
+		// generate color from result with valid alpha bits
+		Color resultWithAlpha = new Color( ColorUtils.scaleAlpha( input.getRGB(), ratio ), true );
+		assertEquals( 0, resultWithAlpha.getRed() );
+		assertEquals( 0, resultWithAlpha.getGreen() );
+		assertEquals( 0, resultWithAlpha.getBlue() );
+
+		assertEquals( 255, input.getAlpha() );
+		// alpha bits are set
+		assertEquals( 127, resultWithAlpha.getAlpha() );
+
+		// generate color from result without valid alpha bits
+		Color resultWithoutAlpha = new Color( ColorUtils.scaleAlpha( input.getRGB(), ratio ) );
+		assertEquals( 0, resultWithoutAlpha.getRed() );
+		assertEquals( 0, resultWithoutAlpha.getGreen() );
+		assertEquals( 0, resultWithoutAlpha.getBlue() );
+
+		// alpha bits are not set -> alpha is still (but wrongly) 255
+		assertEquals( 255, resultWithoutAlpha.getAlpha() );
```

Comment:
I removed the lines myself, and found another similar bug in `ColorUtils.mixColors(...)`.

Now there's a test to covers the alpha channel mixing in `ColorUtils.mixColors(...)`

Reply:
I was kind of anticipating that you judge this code as being misplaced here. For this reason, I had put it in a separate commit ([ColorUtils.mixColors](https://github.com/mastodon-sc/mastodon/pull/231/commits/119c095d02459ca6b86c2e2cfba76fe237a03e11)), so that this commit could just be dropped.

However: I would like to keep this code somewhere, as it contributed very much to solving the bug related to the scaleAlpha() method. It also demonstrates how the result of scaleAlpha should be used, but I confess the unit test, may not be the best place for this. Maybe an improved javadoc would also already help.